templates/commands/render/action_foreach.go

```diff
@@ -42,7 +42,7 @@ func actionForEach(ctx context.Context, fe *spec.ForEach, sp *stepParams) error
 
 	for _, keyVal := range values {
 		subStepParams := sp.WithScope(map[string]string{key: keyVal})
-		if err := executeSteps(ctx, fe.Steps, subStepParams); err != nil {
+		if err := executeSteps(ctx, fe.Steps, subStepParams, ""); err != nil {
```

Comment:
Should we pass the debugDir to substeps like in here? if yes, it will create commits for each substep in addition to the parent step, maybe not necessary.

Reply:
Oh, good point! We should move the `git init` outside of executeSteps.

Reply:
Done!

Reply:
Changed back to have separate commits for substeps, one concern is the commit message does not have all the info (i.e.: parent step), see example:
<img width="714" alt="Screenshot 2024-01-02 at 4 40 51 PM" src="https://github.com/abcxyz/abc/assets/51539171/b12af224-ca80-479d-bed1-6a1acc780d09">

I thought about making step a linked list, so that we can get the info of all parent steps (we will need to pass step around instead of step.ForEach), but that could be a separate topic.

Please let me know if you have any better ideas!

Reply:
For now, how about we just omit the step number from the commit message and call it good?